js/ui/save-html.js

```diff
@@ -31,15 +31,19 @@ define(
                     ,   textAlign:      "center"
                     ,   fontSize:       "inherit"
                     }
-                ,   addButton = function (title, content, fileName, popupContent) {
+                ,   addButton = function (title, href, fileName, popupContent, type) {
+                        if(!type){
+                            href = "data:text/html;charset=utf-8," + encodeURIComponent(href)
+                       }
                         if (supportsDownload) {
                             $("<a></a>")
                                 .appendTo($div)
                                 .text(title)
                                 .css(buttonCSS)
                                 .attr({
-                                    href:   "data:text/html;charset=utf-8," + encodeURIComponent(content)
-                                ,   download:   fileName
+                                    href: href
+                                ,   download: fileName
+                                ,   type: type
```

Comment:
> On 08 Sep 2015, at 10:47 , Tobie Langel <notifications@github.com> wrote:
> 
> In js/ui/save-html.js <https://github.com/w3c/respec/pull/487#discussion_r38902443>:
> 
> >                          if (supportsDownload) {
> >                              $("<a></a>")
> >                                  .appendTo($div)
> >                                  .text(title)
> >                                  .css(buttonCSS)
> >                                  .attr({
> > -                                    href:   "data:text/html;charset=utf-8," + encodeURIComponent(content)
> > -                                ,   download:   fileName
> > +                                    href: href
> > +                                ,   download: fileName
> > +                                ,   type: type
> Feels slightly weird to have this undefined type here (unsure what jquery does with it).
> 
> 

That is not my code… I just added the EPUB access, trying to navigate in the unknown waters of Javascript+respec:-). And this was already there.

Ivan